### PR TITLE
Move ref vs retain when stealing pointer for structured args.

### LINKED ISSUE #9936: TF integrations Python + Vulkan failures
### Issue body

In #9927, I'm consistently hitting the following error in 25 vulkan TF integrations tests

> python3: /home/runner/actions-runner/_work/iree/iree/third_party/vulkan_memory_allocator/include/vk_mem_alloc.h:11939: void VmaDeviceMemoryBlock::Destroy(VmaAllocator): Assertion `m_pMetadata->IsEmpty() && "Some allocations were not freed before destruction of this memory block!"' failed.

This is the same error message as #3381, but @stellaraccident suggested it was probably a different issue. Odd about this issue is that it happens consistently on the GitHub actions runner, but wasn't happening on Kokoro at all. I've tried reducing test parallelism to ensure this isn't some issue of competing for resources (one difference with the new machines is that they have more cores and #9905 hit this issue), but that did not help.

For now, I'm going to disable the Vulkan tests in this job, but this will block #9855.

These are the failing tests:

```
iree_tf_tests/layers/vulkan__Add.run
iree_tf_tests/layers/vulkan__AdditiveAttention.run
iree_tf_tests/layers/vulkan__Attention.run
iree_tf_tests/layers/vulkan__Average.run
iree_tf_tests/layers/vulkan__Concatenate.run
iree_tf_tests/layers/vulkan__Dot.run
iree_tf_tests/layers/vulkan__Maximum.run
iree_tf_tests/layers/vulkan__Minimum.run
iree_tf_tests/layers/vulkan__MultiHeadAttention.run
iree_tf_tests/layers/vulkan__Multiply.run
iree_tf_tests/layers/vulkan__Subtract.run
iree_tf_tests/layers/vulkan__dynamic_dims_Add.run
iree_tf_tests/layers/vulkan__dynamic_dims_Average.run
iree_tf_tests/layers/vulkan__dynamic_dims_Maximum.run
iree_tf_tests/layers/vulkan__dynamic_dims_Minimum.run
iree_tf_tests/layers/vulkan__dynamic_dims_Multiply.run
iree_tf_tests/layers/vulkan__dynamic_dims_Subtract.run
iree_tf_tests/layers/vulkan__full_api_AdditiveAttention.run
iree_tf_tests/layers/vulkan__full_api_Attention.run
iree_tf_tests/layers/vulkan__full_api_Concatenate.run
iree_tf_tests/math/vulkan__accumulate_n.run
iree_tf_tests/math/vulkan__add_n.run
iree_tf_tests/math/vulkan__dynamic_dim_polyval.run
iree_tf_tests/math/vulkan__polyval.run
iree_tf_tests/uncategorized/vulkan__pytree.run
```

Full logs: https://gist.githubusercontent.com/GMNGeoffrey/6e1ca63782858a2d9ee36157351032a1/raw/e7454d9b4a2fbc8dbdd65630b8bd5c145fdeffdd/logs.txt

### PR DESCRIPTION
**Pull request by stellaraccident**
This was a bit tricky to track down. It only affects structured argument
packing/unpacking. We were leaking a reference to the sub-list, which,
given that it usually contains buffers, was causing a leak of any
contained device backed memory buffers. Then much later, with asserts
enabled, the vulkan driver would notice that not all allocated memory
was freed and assert.

I am not completely sure on the cause of the non-determinism. Since the
problem manifests on shutdown, I'm (just guessing) that the normal
shutdown process may be bypassed sometimes? This does only affect a
subset of TF-specific invocation schemes, and until that was obvious, it
was difficult to see the pattern.

Fixes #9936

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
Thanks! Can you try patching in the changes from #9954 to confirm this fixes that issue? It's just a one-line change to the CI to enable these tests.

**Comment by stellaraccident** [comment]
> Thanks! Can you try patching in the changes from #9954 to confirm this fixes that issue? It's just a one-line change to the CI to enable these tests.

Done, but given that I timed this to let me leave precisely at 5pm on Friday... if it doesn't actually work, I will cry.

File: runtime/bindings/python/invoke.cc [hunk 1]
```diff
@@ -452,8 +452,7 @@ class InvokeStatics {
         sub_packer(c, item_list.raw_ptr(), py_item);
       }
       // Push the sub list.
-      iree_vm_ref_t retained =
-          iree_vm_list_retain_ref(item_list.steal_raw_ptr());
+      iree_vm_ref_t retained = iree_vm_list_move_ref(item_list.steal_raw_ptr());
```
Comment (GMNGeoffrey) [inline runtime/bindings/python/invoke.cc:455]: Should this still be called "retained"?

**Review by GMNGeoffrey — APPROVED** [review]
> > Thanks! Can you try patching in the changes from #9954 to confirm this fixes that issue? It's just a one-line change to the CI to enable these tests.
> 
> Done, but given that I timed this to let me leave precisely at 5pm on Friday... if it doesn't actually work, I will cry.

It works! 🙂 Approving because this fixes the bug, but maybe @benvanik wants to review the substance of this

File: runtime/bindings/python/invoke.cc (hunk 1 above)
Comment (stellaraccident) [reply runtime/bindings/python/invoke.cc:455]: Yes, it is still "retaining" its reference. It is just the reference that it stole. The intent with the naming is that it is obvious that someone needs to transfer this or release it.

**Review by benvanik — APPROVED** [review]

**Pull request merged**